src/sampler/record.rs

```diff
@@ -0,0 +1,451 @@
+use anyhow::{Context, Error, Result};
+use std::collections::HashSet;
+use std::fs::File;
+use std::path::PathBuf;
+use std::sync::atomic::{AtomicBool, AtomicUsize, Ordering};
+use std::sync::mpsc::{channel, sync_channel, Receiver, SyncSender};
+use std::sync::Arc;
+use std::time::{Duration, Instant};
+#[cfg(windows)]
+use winapi::um::timeapi;
+
+use crate::core::initialize::initialize;
+use crate::core::types::{MemoryCopyError, Pid, Process, ProcessRetry, StackTrace};
+use crate::storage::Store;
+use crate::ui::summary;
+
+const BILLION: u64 = 1000 * 1000 * 1000; // for nanosleep
+
+// This SampleTime struct helps us sample on a regular schedule ("exactly" 100 times per second, if
+// the sample rate is 100).
+// What we do is -- when doing the 1234th sample, we calculate the exact time the 1234th sample
+// should happen at, which is (start time + nanos_between_samples * 1234) and then sleep until that
+// time
+struct SampleTime {
+    start_time: Instant,
+    nanos_between_samples: u64,
```

Comment:
NIT
Why unsigned? Signed variables are generally better handled by the compiler as the compiler can do more assumptions thus allowing for more optimization. I would prefer unsigned only in places where we deal with binary content rather than the value itself.
I guess that the reasoning here is that there is a need to have huge range as we are dealing with nanoseconds, but int64 is easily able to hold values of many years in nanoseconds.
http://blog.llvm.org/2011/05/what-every-c-programmer-should-know.html#signed_overflow

Reply:
This is old code that got moved, so I'm not sure. It might have been an arbitrary choice.